packages/hardhat-viem/src/internal/errors.ts

```diff
@@ -44,3 +44,21 @@ You can find a list of supported networks here: https://viem.sh/docs/clients/cha
     );
   }
 }
+
+export class DefaultWalletClientNotFoundError extends HardhatViemError {
+  constructor() {
+    super(
+      `Default wallet client not found. You can set the wallet client by passing it as a parameter in the deployContract function:
+
+const walletClient = await hre.viem.getWalletClient(address);
+const contractA = await hre.viem.deployContract("A", [], { walletClient });
+const contractB = await hre.viem.getContractAt("B", address, { walletClient });`
+    );
+  }
+}
+
+export class DeployContractError extends HardhatViemError {
```

Comment:
Let's use a more specific error maybe? With a message like:

```
The deployment transaction '${txHash}` was mined in block '${blockNumber}' but its receipt doesn't contain a contract address
```

Otherwise, if someone runs into this (unlikely) scenario, it will be super hard to figure out what's going on.

Wdyt?